core/trino-main/src/main/java/io/trino/server/testing/TestingTrinoServer.java

```diff
@@ -231,6 +231,8 @@ private TestingTrinoServer(
                 // Use task.writer-count > 1, as this allows to expose writer-concurrency related bugs.
                 .put("task.writer-count", "2")
                 .put("exchange.client-threads", "4")
+                // Reduce memory footprint in tests
+                .put("exchange.max-buffer-size", "4MB")
```

Comment:
`StreamingDirectExchangeBuffer#bufferedPages` is accounted for, so it's odd that it might cause JVM OOM. I guess other allocations might not be accounted for

Reply:
In tests, there are many allocations that are not accounted for.

**For example**, https://github.com/trinodb/tpch/blob/master/src/main/java/io/trino/tpch/TextPool.java#L27 holds a 300 MB byte[] array that is not accounted for.

Also Trino isn't really designed to run on smart watch-sized machines, or to run multiple Trino instances within single JVM, or to run other stuff alongside TrinoServer within same JVM. We do all of these in tests.